Lombiq.Tests.UI/Models/MemoryApplicationLog.cs

```diff
@@ -0,0 +1,50 @@
+using Lombiq.Tests.UI.Services;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace Lombiq.Tests.UI.Models;
+
+/// <summary>
+/// An <see cref="IApplicationLog"/> implementation where the entries are stored in memory.
+/// </summary>
+public record MemoryApplicationLog(string Name, IList<IApplicationLogEntry> Entries)
```

Comment:
Why is this necessary, exactly? Is multiple `await`s really that costly to warrant copying the log into this type? Because keep in mind, that by default, the logs will be in-memory already with `FakeLogger` (see from `FakeLoggerLogApplicationLog`).

Reply:
How does it make easier to debug?

Reply:
Since the logs is converted into this type, you can just explore the  `logs` in `AssertLogsAsync`, and easily do breakpoint conditions since the log entries are synchronously accessible.

Also going back to the fact that we filter the logs in `MemoryApplicationLog.FromLogAsync` so the Shouldly user message will only contain the the offending log entries that cause the assertion failure. This is especially relevant for tests with a lot of filtered out logs, such is the case with security tests. If we already load in the logs to make that filtering, why not keep the results so we don't have to do it again to generate the formatted string?

Reply:
OK, let's keep it, but please add the rationale to the class docs, because currently it seems that it's for perf.

> the Shouldly user message will only contain the the offending log entries that cause the assertion failure.

Yes, and that I'm definitely for, but it doesn't necessarily need this type.

Reply:
> Yes, and that I'm definitely for, but it doesn't necessarily need this type.

I wouldn't say "need", but it makes things more organized and easier to reason about.